File: Python/RECURSION/04-factorials_upto_n.py
class Solution:
    # logic implementation section
    def factorialNumbers(self, N):
        v = []
        fact = 1
        def Factorial(N, i):
            nonlocal fact
            fact *= i
            if fact > N:
                return
            v.append(fact)
            Factorial(N, i+1)
            
        Factorial(N, 1)
        return v

if __name__ == "__main__":
    # main section
    cases = int(input())
    for t in range(cases):
        N=int(input())
        
        ob=Solution()
        ans = ob.factorialNumbers(N)
        for i in ans:
            print(i, end = " ")
        print()

File: C++/RECURSION/04-factorials_upto_n.cpp
#include <bits/stdc++.h>
using namespace std;

#define vi vector<int>
#define lli long long int
#define vlli vector<long long int>
#define ll long long
#define vll vector<long long>

class Solution {
    public:
    vll v;
    long long int fact = 1;
    void Factorial(ll N, ll i) {
        fact *= i;
        if(fact > N) return;
        v.push_back(fact);
        Factorial(N, i+1);
    }
    vll factorialNumbers(ll N) {
        Factorial(N, 1);
        return v;
    }
};

int main() {
    ios_base::sync_with_stdio(false);

    // main section 
    int t = 1;
    cin >> t;
    while(t--) {
        ll N;
        cin >> N;
        Solution obj;
        vll ans = obj.factorialNumbers(N);
        for(auto num : ans) {
            cout << num << " ";
        }
        cout << endl;
    };
    return 0;
}
